openwisp_controller/config/admin.py

```diff
@@ -1362,6 +1362,12 @@ class ConfigSettingsInline(admin.StackedInline):
         model = OrganizationConfigSettings
         form = ConfigSettingsForm
 
+        def get_fields(self, request, obj=None):
+            fields = super().get_fields(request, obj)
+            if not app_settings.WHOIS_CONFIGURED:
+                fields.remove("whois_enabled")
+            return fields
```

Comment:
What if we define the list of fields conditionally in the class instead?

Pseudo-code:
```
fields = []
if settings.registration_enabled:
  fields += ['reg_enabled', 'shared_secret']
if settings.whois_configured:
  fields += ['whois_enabled']
```

Reply:
We can do this, but i am not sure how we can test this as this will be evaluated at import time and thus, overriding settings in tests have no effect.

Reply:
> We can do this, but i am not sure how we can test this as this will be evaluated at import time and thus, overriding settings in tests have no effect.

Good point!